bin/root-transition.sh

```diff
@@ -15,122 +15,124 @@
 # limitations under the License.
 
 # This script extends the lifetime of the self-signed Citadel root certificate in the current cluster.
-# It offers the following options:
-# - check: Check the current root certificate lifetime.
-# - transition: Extend the lifetime of the current root certificate.
-# - verify: Check the new workload certificates are generated.
 # This script requires openssl, kubectl and bc.
 
 trustdomain() {
-  # shellcheck disable=SC2086
-  openssl x509 -in ${1} -noout -issuer | cut -f3 -d'='
+  openssl x509 -in "$1" -noout -issuer | cut -f3 -d'='
 }
 
 check_secret () {
-	# shellcheck disable=SC2006
-	# shellcheck disable=SC2086
-	MD5=`kubectl get secret $1 -o yaml -n $2 | sed -n 's/^.*root-cert.pem: //p' | md5sum | awk '{print $1}'`
-	if [ "$ROOT_CERT_MD5" != "$MD5" ]; then
+  local MD5
+  MD5=$(kubectl get secret $1 -o yaml -n $2 | sed -n \
+    's/^.*root-cert.pem: //p' | md5sum | awk '{print $1}')
```

Comment:
You can define a local variable and initialize it in the same statement (also it's kind of a shell convention to use lower cases for local variables). e.g.: `local md5=$(.......)`

Also pay attention to quotes around variables, lint is complaining about those.

Reply:
I will make the local variables lower cases.
SC2155 encourages to declare and assign variables separately to avoid masking return values:
https://github.com/koalaman/shellcheck/wiki/SC2155